app/helpers/link_to_helper.rb

```diff
@@ -294,4 +292,9 @@ def message_path(message, options = {})
     message_url(message, options.merge(:only_path => true))
   end
 
+  def sanitize_params(raw_params)
+    unsafe_keys = %w[protocol host]
+    raw_params.reject { |k| unsafe_keys.include?(k) }.permit!
+  end
```

Comment:
FWIW - I am not sure we can be 100% sure this returns parameters which have been sanitised in all cases. It works in `current_path_with_locale` because we're using the `path_only` option.

Reply:
Fair point - although I've seen this [done elsewhere, but with `port` excluded as well](https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/18241)

May try to work out what the actual valid search params are and spend a bit more time trying to permit them. Came unstuck with `latest_status` for reasons which escaped me at the time but ... maybe something more like ...

```rb
allowed_keys = %w[query latest_status view combined only_path controller action page]
diff_keys = routing_info.keys - allowed_keys
routing_info.reject { |k| diff_keys.include?(k) }.permit!
```

Sample params:
```
<ActionController::Parameters {
  "query"=>"bob",
  "latest_status"=>["successful"],
  "controller"=>"general",
  "action"=>"search",
  "combined"=>"bob/requests/relevant",
  "only_path"=>true,
  "view"=>nil} permitted: false>
```